master/internal/db/postgres_runs.go

```diff
@@ -0,0 +1,96 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/uptrace/bun"
+
+	"github.com/determined-ai/determined/master/pkg/model"
+)
+
+// updateRunMetadata is a helper function that returns the closure to update the metadata of a run.
+func updateRunMetadata(
+	runID int,
+	rawMetadata map[string]any,
+	flatMetadata []model.RunMetadataIndex,
+	result *map[string]any,
+) func(context.Context, bun.Tx) error {
+	return func(ctx context.Context, tx bun.Tx) error {
```

Comment:
why are we returning this function verse just putting it the code directly in ```UpdateRunMetadata```

Reply:
Good point, I had done this to separate out the transaction logic; however, looking at it again it feels like this introduce unnecessary complexity